File: 2023/D01.kt
package y2023

import java.io.File

fun d01(input: List<String>): Pair<Int, Int> {
    val digits1 = input.map { it.toCharArray().filter { c -> c.isDigit() } }
    val result1 = digits1.sumOf { listOf(it.first(), it.last()).joinToString("").toInt() }

    val input2 = input.map {
        it.replace("one", "one1one")
            .replace("two", "two2two")
            .replace("three", "three3three")
            .replace("four", "four4four")
            .replace("five", "five5five")
            .replace("six", "six6six")
            .replace("seven", "seven7seven")
            .replace("eight", "eight8eight")
            .replace("nine", "nine9nine")
    }
    val digits2 = input2.map { it.toCharArray().filter { c -> c.isDigit() } }
    val result2 = digits2.sumOf { listOf(it.first(), it.last()).joinToString("").toInt() }
    return Pair(result1, result2)
}


fun main() {
    val input = File("inputs/2023/D01.txt").readLines()
    println(d01(input))
}

File: 2023/D01.jl
input = split.(readlines("D01.txt"), ' ')

function D01(input)
    result1 = result2 = 0
    for line in input
        numbers = [filter(isdigit, collect(s)) for s in line]
        result1 += parse(Int, join([numbers[1][1], numbers[1][end]]))

        line2 = replace(line[1], "one" => "one1one")
        line2 = replace(line2, "two" => "two2two")
        line2 = replace(line2, "three" => "three3three")
        line2 = replace(line2, "four" => "four4four")
        line2 = replace(line2, "five" => "five5five")
        line2 = replace(line2, "six" => "six6six")
        line2 = replace(line2, "seven" => "seven7seven")
        line2 = replace(line2, "eight" => "eight8eight")
        line2 = replace(line2, "nine" => "nine9nine")
        numbers = [filter(isdigit, collect(s)) for s in [line2]]
        result2 += parse(Int, join([numbers[1][1], numbers[1][end]]))
    end
    (result1, result2)
end

println(D01(input))